pandas/core/indexes/base.py

```diff
@@ -160,6 +160,12 @@ def _new_Index(cls, d):
         from pandas.core.indexes.period import _new_PeriodIndex
 
         return _new_PeriodIndex(cls, **d)
+
+    if issubclass(cls, ABCMultiIndex):
+        if "labels" in d and "codes" not in d:
+            # GH#23752 "labels" kwarg has been replaced with "codes"
```

Comment:
do we have an issue about removing *this* code at some point, what are the pre-conditions for that?

Reply:
> what are the pre-conditions for that?

not wanting/needing to support sufficiently-old pickles

Reply:
can you clarify, meaning < 0.23.4? later? (if you can pls put in the issue)